nixops/ssh_util.py

```diff
@@ -285,37 +298,44 @@ def split_openssh_args(self, args: Iterable[str]) -> Tuple[List[str], Command]:
                 break
         return (flags, command)
 
-    def _sanitize_command(
-        self, command: Command, allow_ssh_args: bool
+    def _format_command(
+        self, command: Command, user: str, allow_ssh_args: bool,
     ) -> Iterable[str]:
         """
         Helper method for run_command, which essentially prepares and properly
         escape the command. See run_command() for further description.
         """
+
+        # Dont make assumptions about remote login shell
+        cmd: List[str] = ["bash", "-c"]
+
         if isinstance(command, str):
             if allow_ssh_args:
                 return shlex.split(command)
             else:
-                return ["--", command]
+                cmd.append(command)
         # iterable
         elif allow_ssh_args:
             return command
         else:
-            return [
-                "--",
-                " ".join(
-                    ["'{0}'".format(arg.replace("'", r"'\''")) for arg in command]
-                ),
-            ]
+            cmd.append(
+                " ".join(["'{0}'".format(arg.replace("'", r"'\''")) for arg in command])
+            )
+
+        if user and user != "root":
+            cmd.insert(0, "sudo")
```

Comment:
```suggestion
            cmd.insert(0, "sudo")
```

We may want to pass some more options:

```
  -n, --non-interactive         non-interactive mode, no prompts are used
  -i, --login                   run login shell as the target user; a command may also be specified
  -H, --set-home                set HOME variable to target user's home dir
```

though I'm not sure. I'm sort of thinking this implies we should make this configurable right out of the gate?

Reply:
`-n` breaks my use case (SSH agent forwarding + PAM) despite not actually requiring a prompt, so I'd rather not set that as a default
`-i` is a bad idea because it's not the default when using OpenSSH & specifying a command, so it wouldn't be equivalent to a SSH session as root.
`-H` is a good idea. It's already default on my systems but depends on the remote host configuration so it's a good default.